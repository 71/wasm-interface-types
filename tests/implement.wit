;; no-validate

(module
  (import "a" "b" (func $core))
  (import "b" "c" (func))
  (import "b" "c" (func $bar))

  (@interface implement (import "a" "b") (func 100))
  (@interface implement (func $bar) (func 100))
  (@interface implement (func 0) (func 100))

  (@interface implement (func $core))
  (@interface implement (func 0) (param s32))
)

(; CHECK-ALL:
  (module
    (type (;0;) (func))
    (import "a" "b" (func $core (type 0)))
    (import "b" "c" (func (;1;) (type 0)))
    (import "b" "c" (func $bar (type 0)))
    (@interface type (func))
    (@interface type (func (param s32)))
    (@interface func (type 0))
    (@interface func (type 1))
    (@interface implement (func $core) (func 100))
    (@interface implement (func $bar) (func 100))
    (@interface implement (func $core) (func 100))
    (@interface implement (func $core) (func 0))
    (@interface implement (func $core) (func 1)))
;)
