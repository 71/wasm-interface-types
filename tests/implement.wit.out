(module
  (type (;0;) (func))
  (import "a" "b" (func $core (type 0)))
  (import "b" "c" (func (;1;) (type 0)))
  (import "b" "c" (func $bar (type 0)))
  (@interface type (func))
  (@interface type (func (param s32)))
  (@interface func (type 0))
  (@interface func (type 1))
  (@interface implement (func $core) (func 100))
  (@interface implement (func $bar) (func 100))
  (@interface implement (func $core) (func 100))
  (@interface implement (func $core) (func 0))
  (@interface implement (func $core) (func 1)))